modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -130,6 +130,160 @@ Utilities::getProjectionModel(const ComplexData* data,
+                    + " bytes, needed " 
+                    + str::toString(requiredBufferBytes) 
+                    + " bytes"));
+    }
+    else if(bufferNumBytes && !buffer)
+    {
+        // We were provided a buffer size but for some reason
+        // the buffer is still null.
+        
+        throw except::Exception(Ctxt("Null buffer provided to getWidebandData"
+                    + std::string(" when a ")
+                    + str::toString(bufferNumBytes)
+                    + std::string(" byte buffer was expected")));
+    }
+    else if(bufferNumBytes && buffer)
+    {
+        // We have buffer size AND buffer
+        returnBuffer = buffer;
+    }
+    else
+    {
+        // We do not have a buffer size, we need to make our own
+        returnBuffer = new std::complex<float>[numRows*numCols];
+    }
+
+    if(pixelType == PixelType::RE32F_IM32F)
+    {
+        six::Region region;
+        region.setStartRow(startRow);
+        region.setStartCol(startCol);
+        region.setNumRows(numRows);
+        region.setNumCols(numCols);
+        region.setBuffer(reinterpret_cast<UByte*>(returnBuffer));
+        reader.interleaved(region, imageNumber);
+    }
+    else if(pixelType == PixelType::RE16I_IM16I)
+    {
+        const size_t elementsPerRow = numCols*2;
+        const size_t rowsAtATime = 
+            (32000000 / (elementsPerRow * sizeof(short))) + 1;
+        std::vector<short> tempVector(elementsPerRow * rowsAtATime);
+        short* const tempBuffer = &tempVector[0];
+
+        for(size_t row = startRow, rowsToRead = rowsAtATime; 
+                row < numRows; 
+                row += rowsToRead)
+        {
+            if(row + rowsToRead > numRows)
+            {
+                rowsToRead = numRows - row;
+            }
+
+            six::Region region;
+            region.setStartRow(row);
+            region.setStartCol(startCol);
+            region.setNumRows(rowsToRead);
+            region.setNumCols(numCols);
+            region.setBuffer(reinterpret_cast<UByte*>(tempBuffer));
+            reader.interleaved(region, imageNumber);
+
+            //Take each Int16 out of the temp buffer and put it into the real buffer as a Float32
+            float* bufferPtr = reinterpret_cast<float*>(returnBuffer) + (row * elementsPerRow);
+            for(size_t index = 0; index < elementsPerRow * rowsToRead; index++)
+            {
+                bufferPtr[index] = tempBuffer[index];
+            }
+        }
+    }
+    else
+    {
+        throw except::Exception(Ctxt(complexData.getName() + " has an unknown pixel type"));
+    }
+
+    return returnBuffer;
+}
+
+void Utilities::getWidebandData(NITFReadControl& reader,
+                                const ComplexData& complexData,
+                                std::vector<std::complex<float> >& buffer)
+{
+    size_t requiredNumElements = complexData.getNumCols() * complexData.getNumRows();
+    buffer.resize(requiredNumElements);
```

Comment:
To be pedantic, need to handle the case where requiredNumElements is 0 in which case `&buffer[0]` is not valid to do.  Easiest thing is probably just to only call `getWidebandData()` if `!buffer.empty()`.  Also, don't bother with the extra parens - just use `&buffer[0]` rather than `&(buffer[0])`.